tests/adapters/test_json_adapter.py

```diff
@@ -641,6 +641,48 @@ class TestSignature(dspy.Signature):
         _, second_call_kwargs = mock_completion.call_args_list[1]
         assert second_call_kwargs.get("response_format") == {"type": "json_object"}
 
+def test_json_adapter_json_mode_no_structured_outputs():
+    class TestSignature(dspy.Signature):
+        question: str = dspy.InputField()
+        answer: str = dspy.OutputField(desc="String output field")
+
+    dspy.configure(lm=dspy.LM(model="azure_ai/grok-3", cache=False), adapter=dspy.JSONAdapter())
```

Comment:
I am surprised that you found an LM that doesn't support `response_format`... but let's mock `supports_response_schema` so that readers in the future know what clearly what this tests.